File: app/src/main/java/vn/loitp/app/activity/customviews/recyclerview/parallaxrecyclerview/ParallaxAdapter.java
package vn.loitp.app.activity.customviews.recyclerview.parallaxrecyclerview;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.recyclerview.widget.RecyclerView;

import org.jetbrains.annotations.NotNull;

import vn.loitp.app.R;

public class ParallaxAdapter extends RecyclerView.Adapter {

    private Context context;

    ParallaxAdapter(Context context) {
        this.context = context;
    }

    @NotNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NotNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.row_parallax, parent, false);
        return new ParallaxHolder(view);
    }

    @Override
    public void onBindViewHolder(@NotNull RecyclerView.ViewHolder holder, int position) {
    }

    @Override
    public int getItemCount() {
        return 50;
    }

    private static class ParallaxHolder extends RecyclerView.ViewHolder {
        ParallaxHolder(View itemView) {
            super(itemView);
        }
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/recyclerview/parallaxrecyclerview/ParallaxAdapter.kt
package vn.loitp.app.activity.customviews.recyclerview.parallaxrecyclerview

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.core.adapter.AnimationAdapter
import vn.loitp.app.R

class ParallaxAdapter internal constructor(private val context: Context) : AnimationAdapter() {

    override fun onCreateViewHolder(viewGroup: ViewGroup, position: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(context).inflate(R.layout.row_parallax, viewGroup, false))
    }

    override fun getItemCount(): Int {
        return 50
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        if (holder is ViewHolder) {
            holder.bind()
        }
    }

    inner class ViewHolder(v: View) : RecyclerView.ViewHolder(v) {
        fun bind() {
            //do sth
        }
    }

}
